packages/protocol/test-sol/common/FeeHandler.t.sol

```diff
@@ -0,0 +1,951 @@
+    feeHandler.setBurnFraction(100);
+  }
+
+  function test_Reverts_WhenFractionsGreaterThanOne() public {
+    vm.expectRevert("Burn fraction must be less than or equal to 1");
+    feeHandler.setBurnFraction(FixidityLib.newFixedFraction(3, 2).unwrap());
+  }
+
+  function test_SetBurnFraction() public {
+    feeHandler.setBurnFraction(FixidityLib.newFixedFraction(80, 100).unwrap());
+    assertEq(
+      feeHandler.burnFraction(),
+      FixidityLib.newFixedFraction(80, 100).unwrap(),
+      "Burn fraction should be set"
+    );
+  }
+
+  function test_ShouldEmitFeeBeneficiarySet() public {
+    vm.expectEmit(true, true, true, true);
+    emit BurnFractionSet(FixidityLib.newFixedFraction(80, 100).unwrap());
+    feeHandler.setBurnFraction(FixidityLib.newFixedFraction(80, 100).unwrap());
+  }
+}
+
+contract FeeHandlerTest_SetHandler is FeeHandlerTest {
+  function test_Reverts_WhenCallerNotOwner() public {
+    vm.prank(user);
+    vm.expectRevert("Ownable: caller is not the owner");
+    feeHandler.setHandler(address(stableToken), address(mentoSeller));
+  }
+
+  function test_SetsHandler() public {
+    feeHandler.setHandler(address(stableToken), address(mentoSeller));
+    assertEq(
+      feeHandler.getTokenHandler(address(stableToken)),
+      address(mentoSeller),
+      "Handler should be set"
+    );
+  }
+}
+
+contract FeeHandlerTest_AddToken is FeeHandlerTest {
+  function test_Reverts_WhenCallerNotOwner() public {
+    vm.prank(user);
+    vm.expectRevert("Ownable: caller is not the owner");
+    feeHandler.addToken(address(stableToken), address(mentoSeller));
+  }
+
+  function test_AddsToken() public {
+    feeHandler.addToken(address(stableToken), address(mentoSeller));
+    address[] memory expectedActiveTokens = new address[](1);
+    expectedActiveTokens[0] = address(stableToken);
+    assertEq(feeHandler.getActiveTokens(), expectedActiveTokens);
+    assertTrue(feeHandler.getTokenActive(address(stableToken)));
+    assertEq(feeHandler.getTokenHandler(address(stableToken)), address(mentoSeller));
+  }
+
+  function test_ShouldEmitTokenAdded() public {
+    vm.expectEmit(true, true, true, true);
+    emit TokenAdded(address(stableToken), address(mentoSeller));
+    feeHandler.addToken(address(stableToken), address(mentoSeller));
+  }
+}
+
+contract FeeHandlerTest_RemoveToken is FeeHandlerTest {
+  function test_Reverts_WhenCallerNotOwner() public {
+    vm.prank(user);
+    vm.expectRevert("Ownable: caller is not the owner");
+    feeHandler.removeToken(address(stableToken));
+  }
+
+  function test_RemovesToken() public {
+    feeHandler.addToken(address(stableToken), address(mentoSeller));
+    feeHandler.removeToken(address(stableToken));
+    assertFalse(feeHandler.getTokenActive(address(stableToken)));
+    assertEq(feeHandler.getActiveTokens().length, 0);
+    assertEq(feeHandler.getTokenHandler(address(stableToken)), address(0));
+  }
+
+  function test_ShouldEmitTokenRemoved() public {
```

Comment:
Here and throughout. 

I'm bringing these up because of the [WIP PR](https://github.com/celo-org/celo-monorepo/pull/10870) that addresses the naming convention. 

Would not want to have to do the same twice.

```suggestion
  function test_Emits_TokenRemoved() public {
```